### ISSUE TITLE
Lovely doesn't seem to pass on ENV Variables declared in the Application manifest to Helmfile.

### ISSUE THREAD

**Issue by kikokikok** [crumbhole/argocd-lovely-plugin#607]
Hello, 

I am trying to use Lovely to deploy a complex Helmfile based composition
the Application Manifest looks like this

``` yaml
project: default
source:
  repoURL: file:///host/dev/git/app-repo
  path: path_to_helmfile/
  targetRevision: HEAD
  plugin:
    name: lovely-vault-plugin
    env:
      - name: ARGOCD_ENV_CLUSTER_TYPE
        value: cloud
      - name: ARGOCD_ENV_CLUSTER_NAME
        value: k3s
      - name: ARGOCD_ENV_HELMFILE_ENVIRONMENT
        value: k3easy
      - name: ARGOCD_ENV_METRICS_ENABLED
        value: 'true'
      - name: ARGOCD_ENV_CERT_ISSUER_GROUP
        value: certmanager.step.sm
      - name: ARGOCD_ENV_CERT_ISSUER_TYPE
        value: StepClusterIssuer
      - name: ARGOCD_ENV_CERT_ISSUER_NAME
        value: step-cluster-issuer
      - name: ARGOCD_ENV_DEPLOY_AWS_ACCESS_KEY_ID
        value: ref+vault://secret/path/#DEPLOY_AWS_ACCESS_KEY_ID
      - name: ARGOCD_ENV_DEPLOY_AWS_SECRET_ACCESS_KEY
        value: ref+vault://secret/path/#DEPLOY_AWS_SECRET_ACCESS_KEY
destination:
  server: https://kubernetes.default.svc
  namespace: kapp
syncPolicy:
  syncOptions:
    - CreateNamespace=true
    - PruneLast=true
    - Validate=false
```
our helmfile.yaml expects env HELMFILE_ENVIRONMENT in order to select the right template in our gitops convention. I have tried with ou without the ARGOCD_ENV_ Prefix with no success

I always end up with this exception:
```bash
Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = Manifest generation error (cached): plugin sidecar failed. error generating manifests in cmp: rpc error: code = Unknown desc = error generating manifests: `argocd-lovely-plugin` failed exit status 1: 2024/10/28 22:05:53 error running helmfile: exit status 1: WARNING: environments and releases cannot be defined within the same YAML part. Use --- to extract the environments into a dedicated part in ./helmfile.yaml: error during helmfile.yaml.part.0 parsing: template: stringTemplate:7:5: executing "stringTemplate" at <requiredEnv "HELMFILE_ENVIRONMENT">: error calling requiredEnv: required env var `HELMFILE_ENVIRONMENT` is not set
```
Any ideas about what I might be doing wrong or maybe a known limitation ? 

Thanks for the help.

**Comment by Joibel** [comment]
That is correct.

ENV variables are passed to the plugin as parameters, but we don't get access to all the env variables with the current plugin architecture.

You can:
* inject environment variables permanently into the plugin as part of the configuration of its sidecar. I imagine this doesn't work for you if you want to specify environment per app. You can run multiple copies of the plugin, one for each environment, but this may also be suboptimal.
* Use `LOVELY_HELMFILE_TEMPLATE_PARAMS` to pass `--environment` to helmfile template. I imagine this works, but I haven't tried it.
* Craft us a PR for a new parameter containing extra environment variables for this execution of the plugin.

**Comment by Joibel** [comment]
Fixed in #608
